enterprise/api/v1/views/default_enterprise_enrollments.py

```diff
@@ -0,0 +1,275 @@
+
+    @property
+    def requested_enterprise_customer_uuid(self):
+        """
+        Get and validate the enterprise customer UUID from the query parameters.
+        """
+        if not (enterprise_customer_uuid := self.request.query_params.get('enterprise_customer_uuid')):
+            raise ValidationError({"detail": "enterprise_customer_uuid is a required query parameter."})
+
+        try:
+            return UUID(enterprise_customer_uuid)
+        except ValueError as exc:
+            raise ValidationError({
+                "detail": "enterprise_customer_uuid query parameter is not a valid UUID."
+            }) from exc
+
+    @property
+    def requested_lms_user_id(self):
+        """
+        Get the (optional) LMS user ID from the request.
+        """
+        return self.request.query_params.get('lms_user_id')
+
+    @property
+    def base_queryset(self):
+        """
+        Required by the `PermissionRequiredForListingMixin`.
+        For non-list actions, this is what's returned by `get_queryset()`.
+        For list actions, some non-strict subset of this is what's returned by `get_queryset()`.
+        """
+        kwargs = {}
+        if self.requested_enterprise_customer_uuid:
+            kwargs['enterprise_customer'] = self.requested_enterprise_customer_uuid
+        return models.DefaultEnterpriseEnrollmentIntention.objects.filter(**kwargs)
+
+    @property
+    def user_for_learner_status(self):
+        """
+        Get the user for learner status based on the request.
+        """
+        if self.request.user.is_staff and self.requested_lms_user_id is not None:
+            # If the user is staff and a lms_user_id is provided, return the specified user.
+            User = get_user_model()
+            try:
+                return User.objects.get(id=self.requested_lms_user_id)
+            except User.DoesNotExist:
+                return None
+
+        # Otherwise, return the request user.
+        return self.request.user
+
+    def get_permission_object(self):
+        """
+        Used for "retrieve" actions. Determines the context (enterprise UUID) to check
+        against for role-based permissions.
+        """
+        return str(self.requested_enterprise_customer_uuid)
+
+    @action(detail=False, methods=['get'], url_path='learner-status')
+    def learner_status(self, request):
+        """
+        Get the status of the learner's enrollment in the default enterprise course.
+        """
+        # Validate the enterprise customer uuid.
+        try:
+            enterprise_customer_uuid = self.requested_enterprise_customer_uuid
+        except ValidationError as exc:
+            return Response(exc, status=status.HTTP_400_BAD_REQUEST)
+
+        # Validate the user for learner status exists and is associated
+        # with the enterprise customer.
+        if not (user_for_learner_status := self.user_for_learner_status):
```

Comment:
nit: you could make `self.user_for_learner_status` a `cached_property` to avoid storing its value in a variable here.